### PR TITLE
fix: fix web-component tag name transform issue

### PR DESCRIPTION
**Pull request by rqzheng2015**
## Description

Fix the web-component tag issue mentioned in https://github.com/BuilderIO/mitosis/issues/1303

### Issue Reason
The babel  transform will automatically identify the "-" as an operator, and add a space before and after it.
So when it met a tag name such as `swiper-container` provided by `web-components`, it will transform the tag name as `swiper - container`. 

### How to Fix
We need to replace ' - ' with '-'  in the babel transformed code to fix this issue.

### CODE REVIEW

File: packages/core/src/helpers/babel-transform.ts [hunk 1]
```diff
@@ -85,7 +85,13 @@ export const babelTransformCode = <VisitorContextType = any>(
   code: string,
   visitor?: Visitor<VisitorContextType>,
   stripTypes = false,
-) => babelTransform({ code, visitor, stripTypes })?.code || '';
+) => {
+  // Fix web component tag issue due to the `babelTransform` function execution
+  // For exmaple: we pass a tag called "swiper-container", and it will be renamed as "swiper - container" after babel transforming,
+  // because babel will automatically identify the "-" as an operator, and add a space before and after it.
+  // So we need to replace ' - ' with '-' to fix this issue.
+  return babelTransform({ code, visitor, stripTypes })?.code?.replace(' - ', '-') || '';
```
[the remark this answers is not in the document]
Comment (samijaber) [reply packages/core/src/helpers/babel-transform.ts:93]: - If you run `yarns snapupdate` in `mitosis/core`, you will see how much this code change impacts generated code.
- you also need to `yarn run fmt:prettier` this file.

**Comment by samijaber** [comment]
I just commented a better solution to this problem here: https://github.com/BuilderIO/mitosis/issues/1303#issuecomment-1850347789

Thanks a lot for taking this on!

File: packages/core/src/helpers/babel-transform.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (rqzheng2015) [reply packages/core/src/helpers/babel-transform.ts:93]: I see, thanks @samijaber. I will soon submit a new patch for this PR.

**Comment by rqzheng2015** [comment]
Hi @samijaber, I have commited new changes based on your advices, and have added snapshot and unit test. Please hava a look when you have time.

**Comment by rqzheng2015** [comment]
Hi @samijaber! Please check the pr if you have the free time, thanks.

**Review by samijaber — APPROVED** [review]
Wonderful fix. Well done. Thanks for the contribution! 🎉

**Pull request merged**